src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -286,7 +285,7 @@ int sd_dhcp_server_set_next_server(sd_dhcp_server *server, const struct in_addr
 int sd_dhcp_server_set_filename(sd_dhcp_server *server, const char *filename) {
         assert_return(server, -EINVAL);
 
-        if (filename && !ascii_is_valid(filename))
+        if (filename && !string_is_safe(filename))
```

Comment:
I think this should just be `filename && !string_is_safe() or !ascii_is_valid()` instead. Then you don't need to change the meaning of `string_is_safe()`.